src/server.rs

```diff
@@ -299,6 +308,10 @@ async fn process_error(err: Rejection) -> Result<impl Reply, Rejection> {
     }
 }
 
+#[derive(Debug)]
+struct NotUtf8;
+impl warp::reject::Reject for NotUtf8 {}
```

Comment:
Please don't add a new (unnecessary) error type and then also not handle it in `process_error`. I'm pretty sure that this will cause either a `404` or a `500` with no useful information to be returned. Instead, just add another variant to the `Error` enum. The compiler should then force you to handle it properly.

Reply:
Yep, just temporary, I need to use `snafu` errors?

Reply:
Yeah, just copy one of the `Error` variants together with the attribute, change the message, handle the new variant in `process_error` and you should be able to return that variant instead of `NotUtf8`.

Reply:
Yep, I understood now how it works, but using enum and parsing JSON, is interesting, so I will create a new error type after that!